### ISSUE TITLE
Parsl Documentation Organization

### ISSUE THREAD

**Issue by dgasmith** [Parsl/parsl#1331]
When approaching the documentation it takes quite a few clicks to find the area that I am looking for. A recent example was trying to find the description of blocks, I went through tutorial, user guide, FAQ, and reference before starting to search closely through the user guide.

A possible solution to this is to have additional high level categories such as:
 - Composition - Overview/Apps/Futures/Composing
 - Execution/Setup - Data/Execution/Configuration
 - Debugging/Advanced - Everything else

Having some of the topics pre-expanded like [here](http://docs.qcarchive.molssi.org/projects/QCFractal/en/stable/) could be useful as well. Though the example TOC is probably too long!

**Comment by danielskatz** [comment]
Would a better index do what you want?

**Comment by dgasmith** [comment]
Index as in the landing page or index as in expanding the TOC sidebar?

**Comment by danielskatz** [comment]
I was thinking of https://parsl.readthedocs.io/en/latest/genindex.html

**Comment by dgasmith** [comment]
For myself, I don't find genindices very useful.

**Comment by WardLT** [comment]
#3435 is probably a duplicate of this
